zio-kafka/src/main/scala/zio/kafka/utils/SslHelper.scala

```diff
@@ -0,0 +1,64 @@
+package zio.kafka.utils
+
+import zio.{ Task, ZIO }
+
+import java.net.InetSocketAddress
+import java.nio.ByteBuffer
+import java.nio.channels.SocketChannel
+
+object SslHelper {
+  // https://issues.apache.org/jira/browse/KAFKA-4090
+  def validateEndpoint(bootstrapServers: List[String], props: Map[String, AnyRef]): Task[Unit] =
+    ZIO
+      .unless(
+        props
+          .get("security.protocol")
+          .collect { case x: String =>
+            x
+          }
+          .exists(_.toLowerCase().contains("SSL"))
+      ) {
+        ZIO.foreachParDiscard(bootstrapServers) { str =>
+          ZIO.scoped {
+            for {
+              address <- ZIO.attempt {
+                           val arr  = str.split(":")
```

Comment:
🔴 That might be a too bit simplistic parsing mechanism 🤔

What if there's no `port` part, and so no `":"`?
Also, what if there's more than one `":"` in the bootstrap server address? 🤔

Reply:
so, it is a list originally and can't contain a few records in one entry. 

Switched to  ClientUtils.parseAndValidateAddress from kafka

Reply:
Fixed